hvpy/io.py

```diff
@@ -54,4 +53,23 @@ def url(self) -> str:
         """
         Final API endpoint URL.
         """
-        return BASE_URL + self.__class__.__name__[:-15] + "/"
+        base_url = get_api_url()
+        return base_url + self.__class__.__name__[:-15] + "/"
+
+
+def get_api_url() -> str:
+    """
+    Returns the base URL for all API calls.
+    """
+    if "PRIVATE_URL" in os.environ:
+        base_url = os.environ["PRIVATE_URL"]
+        del os.environ["PRIVATE_URL"]
+        return base_url
```

Comment:
I still do not like this.

Reply:
I am sorry for this. But I don't know, how do we use `set_api_url()` to modify the base URL of the `HvpyParameters` class?

Reply:
Please don't apologize. I don't have a good idea at the moment either.

Reply:
Let's make base_url a file level variable `_base_url`

Define it to either `api.helioviewer.org` or whatever is in the environment.

Something like
```python
_base_url = os.environ['PRIVATE_URL'] if 'PRIVATE_URL' in os.environ else "https://api.helioviewer.org/v2/"
# or if you don't like python's strange ternary
_base_url = "https://api.helioviewer.org/v2/"
if 'PRIVATE_URL' in os.environ:
  _base_url = os.environ['PRIVATE_URL']
```

This meets the first requirement of starting with our default or the user's environment.

`set_api_url` modified the file level `_base_url`

`HvpyParameters` uses the file level `_base_url`

That should do it.